### PR TITLE
Add allowlist support to Kritis GAP

### PR DESCRIPTION
**Pull request by ooq**

### CODE REVIEW

**Comment by ooq** [comment]
@aysylu @nenaddedic PTAL, thanks!

**Comment by ooq** [comment]
Since this PR is part of a bundle for compatibility change, it should target a feature branch instead? @aysylu

File: pkg/kritis/review/review.go [hunk 1]
```diff
@@ -52,12 +52,39 @@ func New(client metadata.Fetcher, c *Config) Reviewer {
 	}
 }
 
+func imageInGapAllowlist(image string, gap v1beta1.GenericAttestationPolicy) bool {
+	for _, i := range gap.Spec.ImageAllowlist {
+		if i == image {
```
Comment (nenaddedic) [inline pkg/kritis/review/review.go]: Let's make it match the semantics of the whitelist in Binary Authorization:
https://cloud.google.com/binary-authorization/docs/policy-yaml-reference#admissionwhitelistpatterns

**Comment by aysylu** [comment]
> Since this PR is part of a bundle for compatibility change, it should target a feature branch instead?

It's up to you. We could a) hold off on merging and mark all related PRs with 0.3.0 milestone, or b) do development in a feature branch, c) do development in a fork, do reviews there then merge into master.

**Comment by ooq** [comment]
> > Since this PR is part of a bundle for compatibility change, it should target a feature branch instead?
> 
> It's up to you. We could a) hold off on merging and mark all related PRs with 0.3.0 milestone, or b) do development in a feature branch, c) do development in a fork, do reviews there then merge into master.

Cool. Let's do feature branch in this repo then -- so we can both leverage the existing testing infras, and also merge PRs as we go.

**Comment by ooq** [comment]
Closing this to open another one targeting feature branch.

**Pull request closed without merging**